sdk/core/azure-json/src/main/java/com/azure/json/JsonNumber.java

```diff
@@ -0,0 +1,66 @@
+package com.azure.json;
+
+import java.io.IOException;
+
+/**
+ * Class representing the JSON number type
+ */
+public class JsonNumber extends JsonElement {
+    /**
+     * Stores the String representation of the current state of the JsonNumber
+     * object.
+     */
+    private String numberValue;
```

Comment:
Similar feedback here - probably best to internally represent as a number, rather than as a String